tests/gtest/avifpixitest.cc

```diff
@@ -0,0 +1,107 @@
+// Copyright 2025 Google LLC
+// SPDX-License-Identifier: BSD-2-Clause
+
+#include <fstream>
+
+#include "avif/avif.h"
+#include "aviftest_helpers.h"
+#include "gtest/gtest.h"
+
+using ::testing::Combine;
+using ::testing::Values;
+
+namespace avif {
+namespace {
+
+// Used to pass the data folder path to the GoogleTest suites.
+const char* data_path = nullptr;
+
+//------------------------------------------------------------------------------
+
+#if defined(AVIF_ENABLE_EXPERIMENTAL_EXTENDED_PIXI)
+TEST(AvifPixiTest, SameOutput) {
+  ImagePtr image =
+      testutil::CreateImage(4, 4, 8, AVIF_PIXEL_FORMAT_YUV420, AVIF_PLANES_YUV);
+  ASSERT_NE(image, nullptr);
+  image->yuvChromaSamplePosition = AVIF_CHROMA_SAMPLE_POSITION_VERTICAL;
+  testutil::FillImageGradient(image.get());  // The pixels do not matter.
+
+  // Encode.
+
+  testutil::AvifRwData encoded_regular_pixi;
+  EncoderPtr encoder_regular_pixi(avifEncoderCreate());
+  ASSERT_NE(encoder_regular_pixi, nullptr);
+  encoder_regular_pixi->speed = AVIF_SPEED_FASTEST;
+  encoder_regular_pixi->headerFormat = AVIF_HEADER_FULL;
+  ASSERT_EQ(avifEncoderWrite(encoder_regular_pixi.get(), image.get(),
+                             &encoded_regular_pixi),
+            AVIF_RESULT_OK);
+
+  testutil::AvifRwData encoded_extended_pixi;
+  EncoderPtr encoder_extended_pixi(avifEncoderCreate());
+  ASSERT_NE(encoder_extended_pixi, nullptr);
+  encoder_extended_pixi->speed = AVIF_SPEED_FASTEST;
+  encoder_extended_pixi->headerFormat = AVIF_HEADER_FULL_WITH_EXTENDED_PIXI;
+  ASSERT_EQ(avifEncoderWrite(encoder_extended_pixi.get(), image.get(),
+                             &encoded_extended_pixi),
```

Comment:
maybe check that encoded_extended_pixi is bigger than encoded_regular_pixi to make sure it's doing something?